tests/osaka/eip7692_eof_v1/eip6206_jumpf/test_jumpf_execution.py

```diff
@@ -235,39 +235,64 @@ def test_jumpf_stack_size_1024_at_push(
     )
 
 
+@pytest.mark.parametrize(
+    ("stack_height", "failure"),
+    (
+        pytest.param(1021, False, id="no_overflow"),
+        pytest.param(1022, True, id="rule_overflow"),
+        pytest.param(1023, True, id="execution_overflow"),
+    ),
+)
 def test_jumpf_stack_overflow(
+    stack_height: int,
+    failure: bool,
     eof_state_test: EOFStateTestFiller,
 ):
-    """Test stack overflowing 1024 items in JUMPF target function"""
+    """
+    Test stack overflowing 1024 items in JUMPF target function
+
+    `no_overflow` - the stack does not overflow at JUMPF call, executes to end
+    `rule_overflow` - reserved stack overflows, but execution would not overflow
+    `execution_overflow` - execution would overflow (but still blocked by reserved stack rule)
```

Comment:
I don't quite understand the difference between these two scenarios. What is the reserved stack?

Reply:
Elaborated in the comments.  Without rule #2 in the EIP then the `rule_overflow` case would execute correctly. `execution_overflow` fails without rule #2 as well, but rule #2 kills it first.  No way to distinguish the two failures in-evm however, so we need to test the "rule overflow" case to ensure it's being applied.